engine/texc/src/texc_private.h

```diff
@@ -16,37 +16,67 @@
 #include <dlib/array.h>
 #include <stdlib.h>
 #include <stdint.h>
-#include <PVRTexture.h>
 #include "texc.h"
 
+#include <basis/encoder/basisu_enc.h>
+
 /**
  * Texture processing internal types
  */
 namespace dmTexc
 {
     static const uint32_t COMPRESSION_ENABLED_PIXELCOUNT_THRESHOLD = 64; // do not compress mips with less than this pixelcount
 
+    struct Texture;
+
+    struct Encoder
+    {
+        bool     (*m_FnCreate)(Texture* texture, uint32_t width, uint32_t height, PixelFormat pixel_format, ColorSpace color_space, CompressionType compression_type, void* data);
+        void     (*m_FnDestroy)(Texture* texture);
+        bool     (*m_FnGenMipMaps)(Texture* texture);
+        bool     (*m_FnResize)(Texture* texture, uint32_t width, uint32_t height);
+        bool     (*m_FnEncode)(Texture* texture, int num_threads, PixelFormat pixel_format, CompressionType compression_type, CompressionLevel compression_level);
+        uint32_t (*m_FnGetTotalDataSize)(Texture* texture);
+        uint32_t (*m_FnGetData)(Texture* texture, void* out_data, uint32_t out_data_size);
+        bool     (*m_FnPreMultiplyAlpha)(Texture* texture);
+        bool     (*m_FnFlip)(Texture* texture, FlipAxis flip_axis);
+        bool     (*m_FnGetHeader)(Texture* texture, Header* out_header);
+    };
```

Comment:
A better separation between implementations.
Allows us to more easily add/switch to another one in the future.